### PR TITLE
Update all integration targets to use RelWithDebInfo and increase size of ARM OpenSSH test host

### PR DESCRIPTION
**Pull request by andrewhop**
### Description of changes: 
Integration tests that use a lot of crypto can benefit from a faster AWS-LC build. Also increase the size of the size of the ARM OpenSSH host which has been taking well over an hour to complete. 

Skip `channel-timeout` for OpenSSH which added a new multiplexing test in https://github.com/openssh/openssh-portable/commit/9707c8170c0c1baeb1e06e5a53f604498193885f

Disable [public suffix](https://publicsuffix.org/) support (PSL)  in curl integration tests which became a required option in https://github.com/curl/curl/commit/2998874bb61ac6ef3b72d6a61467cd2aaf6e53ea

### Call-outs:
Some integration tests used the static build and this change maintained that behavior, in a future change all the integrations should be updated to use a shared library build so we can verify AWS-LC is in fact being loaded.

### Testing:
The CI will do it. 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and the ISC license.

### CODE REVIEW

**Review by justsmth — DISMISSED** [review]

File: tests/ci/integration/run_curl_integration.sh [hunk 1]
```diff
@@ -30,14 +30,14 @@ cd ${SCRATCH_FOLDER}
 
 function curl_build() {
   autoreconf -fi
-  ./configure --enable-warnings --enable-werror --with-openssl=${AWS_LC_INSTALL_FOLDER}
-  make -j ${NUM_CPU_THREADS}
-  make -j ${NUM_CPU_THREADS} examples
+  ./configure --enable-warnings --enable-werror --with-openssl=${AWS_LC_INSTALL_FOLDER} --without-libpsl
+  make -j "$NUM_CPU_THREADS"
+  make -j "$NUM_CPU_THREADS" examples
```
Comment (WillChilds-Klein) [inline tests/ci/integration/run_curl_integration.sh:35]: we're not building `.so`s for AWS-LC, so we can't use the `ldd` validation we use elsewhere to assert that the project under test is built properly against AWS-LC.

the `curl` distribution on my ubuntu 22.04 box shows the OPENSSL_VERSION string in its `--version` output. perhaps we could grep for AWS-LC there?

```
$ curl --version
curl 7.81.0 (x86_64-pc-linux-gnu) libcurl/7.81.0 OpenSSL/3.0.2 zlib/1.2.11 brotli/1.0.9 zstd/1.4.8 libidn2/2.3.2 libpsl/0.21.0 (+libidn2/2.3.2) libssh/0.9.6/openssl/zlib nghttp2/1.43.0 librtmp/2.3 OpenLDAP/2.5.14
Release-Date: 2022-01-05
Protocols: dict file ftp ftps gopher gophers http https imap imaps ldap ldaps mqtt pop3 pop3s rtmp rtsp scp sftp smb smbs smtp smtps telnet tftp
Features: alt-svc AsynchDNS brotli GSS-API HSTS HTTP2 HTTPS-proxy IDN IPv6 Kerberos Largefile libz NTLM NTLM_WB PSL SPNEGO SSL TLS-SRP UnixSockets zstd
```

File: tests/ci/integration/run_haproxy_integration.sh [hunk 2]
```diff
@@ -52,14 +52,14 @@ cd haproxy
...
+aws_lc_build "$SRC_ROOT" "$AWS_LC_BUILD_FOLDER" "$AWS_LC_INSTALL_FOLDER" -DBUILD_TESTING=OFF -DBUILD_TOOL=OFF -DCMAKE_BUILD_TYPE=RelWithDebInfo -DBUILD_SHARED_LIBS=0
 build_and_test_haproxy
 
 rm -rf "${AWS_LC_INSTALL_FOLDER:?}"/*
 rm -rf "${AWS_LC_BUILD_FOLDER:?}"/*
 
 # Test with shared AWS-LC libraries
-aws_lc_build "${SRC_ROOT}" "${AWS_LC_BUILD_FOLDER}" "${AWS_LC_INSTALL_FOLDER}" -DBUILD_TESTING=OFF -DBUILD_SHARED_LIBS=1 -DCMAKE_BUILD_TYPE=RelWithDebInfo
+aws_lc_build "$SRC_ROOT" "$AWS_LC_BUILD_FOLDER" "$AWS_LC_INSTALL_FOLDER" -DBUILD_TESTING=OFF -DBUILD_TOOL=OFF -DCMAKE_BUILD_TYPE=RelWithDebInfo -DBUILD_SHARED_LIBS=1
 export LD_LIBRARY_PATH="${LD_LIBRARY_PATH:-}:${AWS_LC_INSTALL_FOLDER}/lib/"
```
Comment (WillChilds-Klein) [inline tests/ci/integration/run_haproxy_integration.sh:63]: similar to above, can we use `ldd` to confirm proper linkage against AWS-LC?

please feel free to tell me to go away if this is out of scope :) perhaps this should be a distinct PR.

Comment (andrewhop) [reply tests/ci/integration/run_haproxy_integration.sh:63]: Yeah a separate PR was my plan, this one was all about CI speed while maintaining the same behavior.

File: tests/ci/integration/run_curl_integration.sh (hunk 1 above)
Comment (andrewhop) [reply tests/ci/integration/run_curl_integration.sh:35]: Will do in a follow up PR

**Review by WillChilds-Klein — APPROVED** [review]

**Review by justsmth — APPROVED** [review]

**Pull request merged**